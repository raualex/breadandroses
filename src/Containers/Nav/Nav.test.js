import React from 'react';
import { Nav, mapDispatchToProps } from './';
import { shallow } from 'enzyme';

describe('Nav', () => {
  let wrapper;
  let mockFetchSenate = jest.fn()
  let mockFetchHouse = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<Nav fetchSenate={mockFetchSenate} fetchHouse={mockFetchHouse} />)
  });
  
  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  describe('mapDispatchToProps function', () => {
    const mockDispatch = jest.fn()
    let mappedProps;

    beforeEach(() => {
      mappedProps = mapDispatchToProps(mockDispatch)
    })

    it('should call dispatch when fetchSenate is called', () => {
      mappedProps.fetchSenate()
      expect(mockDispatch).toHaveBeenCalled()
    })

    it('should call dispatch when fetchHouse is called', () => {
      mappedProps.fetchHouse()
      expect(mockDispatch).toHaveBeenCalled()
    })
  })
})
