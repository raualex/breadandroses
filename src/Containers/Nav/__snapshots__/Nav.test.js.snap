// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    fetchHouse={[MockFunction]}
    fetchSenate={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="nav-title-container"
        >
          <h1
            className="nav-title"
          >
            BREAD
          </h1>
          <h1
            className="nav-title"
          >
            &
          </h1>
          <h1
            className="nav-title"
          >
            ROSES
          </h1>
        </div>,
        <div
          className="btn-container"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="senate-link"
            name="senate"
            onClick={[MockFunction]}
            to="/senate"
          >
            <button
              className="senate-btn"
              name="senate"
            >
              SENATE
            </button>
          </NavLink>
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="house-link"
            name="house"
            onClick={[MockFunction]}
            to="/house"
          >
            <button
              className="house-btn"
              name="house"
            >
              HOUSE
            </button>
          </NavLink>
        </div>,
        <div
          className="nav-star-container"
        >
          <img
            alt="star"
            className="nav-star"
            src="back_star.png"
          />
          <img
            alt="fist"
            className="nav-fist"
            src="power.svg"
          />
        </div>,
      ],
      "className": "header-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="nav-title"
            >
              BREAD
            </h1>,
            <h1
              className="nav-title"
            >
              &
            </h1>,
            <h1
              className="nav-title"
            >
              ROSES
            </h1>,
          ],
          "className": "nav-title-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "BREAD",
              "className": "nav-title",
            },
            "ref": null,
            "rendered": "BREAD",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "&",
              "className": "nav-title",
            },
            "ref": null,
            "rendered": "&",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ROSES",
              "className": "nav-title",
            },
            "ref": null,
            "rendered": "ROSES",
            "type": "h1",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="senate-link"
              name="senate"
              onClick={[MockFunction]}
              to="/senate"
            >
              <button
                className="senate-btn"
                name="senate"
              >
                SENATE
              </button>
            </NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="house-link"
              name="house"
              onClick={[MockFunction]}
              to="/house"
            >
              <button
                className="house-btn"
                name="house"
              >
                HOUSE
              </button>
            </NavLink>,
          ],
          "className": "btn-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": <button
                className="senate-btn"
                name="senate"
              >
                SENATE
              </button>,
              "className": "senate-link",
              "name": "senate",
              "onClick": [MockFunction],
              "to": "/senate",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SENATE",
                "className": "senate-btn",
                "name": "senate",
              },
              "ref": null,
              "rendered": "SENATE",
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": <button
                className="house-btn"
                name="house"
              >
                HOUSE
              </button>,
              "className": "house-link",
              "name": "house",
              "onClick": [MockFunction],
              "to": "/house",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "HOUSE",
                "className": "house-btn",
                "name": "house",
              },
              "ref": null,
              "rendered": "HOUSE",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="star"
              className="nav-star"
              src="back_star.png"
            />,
            <img
              alt="fist"
              className="nav-fist"
              src="power.svg"
            />,
          ],
          "className": "nav-star-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "star",
              "className": "nav-star",
              "src": "back_star.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "fist",
              "className": "nav-fist",
              "src": "power.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="nav-title-container"
          >
            <h1
              className="nav-title"
            >
              BREAD
            </h1>
            <h1
              className="nav-title"
            >
              &
            </h1>
            <h1
              className="nav-title"
            >
              ROSES
            </h1>
          </div>,
          <div
            className="btn-container"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="senate-link"
              name="senate"
              onClick={[MockFunction]}
              to="/senate"
            >
              <button
                className="senate-btn"
                name="senate"
              >
                SENATE
              </button>
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="house-link"
              name="house"
              onClick={[MockFunction]}
              to="/house"
            >
              <button
                className="house-btn"
                name="house"
              >
                HOUSE
              </button>
            </NavLink>
          </div>,
          <div
            className="nav-star-container"
          >
            <img
              alt="star"
              className="nav-star"
              src="back_star.png"
            />
            <img
              alt="fist"
              className="nav-fist"
              src="power.svg"
            />
          </div>,
        ],
        "className": "header-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="nav-title"
              >
                BREAD
              </h1>,
              <h1
                className="nav-title"
              >
                &
              </h1>,
              <h1
                className="nav-title"
              >
                ROSES
              </h1>,
            ],
            "className": "nav-title-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "BREAD",
                "className": "nav-title",
              },
              "ref": null,
              "rendered": "BREAD",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "&",
                "className": "nav-title",
              },
              "ref": null,
              "rendered": "&",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ROSES",
                "className": "nav-title",
              },
              "ref": null,
              "rendered": "ROSES",
              "type": "h1",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="senate-link"
                name="senate"
                onClick={[MockFunction]}
                to="/senate"
              >
                <button
                  className="senate-btn"
                  name="senate"
                >
                  SENATE
                </button>
              </NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="house-link"
                name="house"
                onClick={[MockFunction]}
                to="/house"
              >
                <button
                  className="house-btn"
                  name="house"
                >
                  HOUSE
                </button>
              </NavLink>,
            ],
            "className": "btn-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": <button
                  className="senate-btn"
                  name="senate"
                >
                  SENATE
                </button>,
                "className": "senate-link",
                "name": "senate",
                "onClick": [MockFunction],
                "to": "/senate",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "SENATE",
                  "className": "senate-btn",
                  "name": "senate",
                },
                "ref": null,
                "rendered": "SENATE",
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": <button
                  className="house-btn"
                  name="house"
                >
                  HOUSE
                </button>,
                "className": "house-link",
                "name": "house",
                "onClick": [MockFunction],
                "to": "/house",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "HOUSE",
                  "className": "house-btn",
                  "name": "house",
                },
                "ref": null,
                "rendered": "HOUSE",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="star"
                className="nav-star"
                src="back_star.png"
              />,
              <img
                alt="fist"
                className="nav-fist"
                src="power.svg"
              />,
            ],
            "className": "nav-star-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "star",
                "className": "nav-star",
                "src": "back_star.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "fist",
                "className": "nav-fist",
                "src": "power.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
